packages/@aws-cdk/aws-vpcv2-alpha/lib/vpc-v2.ts

```diff
@@ -0,0 +1,464 @@
+        }
+        //validate CIDR ranges per RFC 1918
+        if (secondaryVpcOptions.ipv4CidrBlock!) {
+          const ret = validateIpv4address(secondaryVpcOptions.ipv4CidrBlock, this.resource.cidrBlock);
+          if (ret === false) {
+            throw new Error('CIDR block should be in the same RFC 1918 range in the VPC');
+          }
+        }
+
+        const cfnVpcCidrBlock = new CfnVPCCidrBlock(this, `SecondaryIp${ipCount}`, {
+          vpcId: this.vpcId,
+          cidrBlock: secondaryVpcOptions.ipv4CidrBlock,
+          ipv4IpamPoolId: secondaryVpcOptions.ipv4IpamPool?.ipamPoolId,
+          ipv4NetmaskLength: secondaryVpcOptions.ipv4NetmaskLength,
+          ipv6NetmaskLength: secondaryVpcOptions.ipv6NetmaskLength,
+          ipv6IpamPoolId: secondaryVpcOptions.ipv6IpamPool?.ipamPoolId,
+          amazonProvidedIpv6CidrBlock: secondaryVpcOptions.amazonProvided,
+        });
+        if (secondaryVpcOptions.dependencies) {
+          for (const dep of secondaryVpcOptions.dependencies) {
+            cfnVpcCidrBlock.addDependency(dep);
+          }
+        }
+        //Create secondary blocks for Ipv4 and Ipv6
+        this.secondaryCidrBlock.push(cfnVpcCidrBlock);
+      }
+    }
+
+    /**
+     * Empty array for isolated subnets
+     */
+    this.isolatedSubnets = new Array<ISubnet>;
+
+    /**
+     * Empty array for public subnets
+     */
+    this.publicSubnets = new Array<ISubnet>;
+
+    /**
+     * Empty array for private subnets
+     */
+    this.privateSubnets = new Array<ISubnet>;
+
+    /**
+     * Dependable that can be depended upon to force internet connectivity established on the VPC
+     * Add igw to this if its a public subnet
+     * */
+    this.internetConnectivityEstablished = this._internetConnectivityEstablished;
+  }
+}
+/**
+ * Supports assigning IPv4 address to VPC
+ */
+class ipv4CidrAllocation implements IIpAddresses {
+
+  constructor(private readonly cidrBlock: string) {
+
+  }
+
+  /**
+   * @returns CIDR block provided by the user to set IPv4
+   */
+  allocateVpcCidr(): VpcCidrOptions {
+    return {
+      ipv4CidrBlock: this.cidrBlock,
+    };
+  }
+}
+
+/**
+ * Supports Amazon Provided Ipv6 ranges
+ */
+class AmazonProvided implements IIpAddresses {
+  /**
+   * Represents an Amazon-provided IPv6 CIDR range for a VPC.
+   *
+   * This class implements the IIpAddresses interface and is used to allocate an Amazon-provided
+   * IPv6 CIDR range for a VPC. When an instance of this class is used to allocate the VPC CIDR,
+   * Amazon will automatically assign an IPv6 CIDR range from its pool of available addresses.
+   */
+
+  constructor() {};
+
+  allocateVpcCidr(): VpcCidrOptions {
+    return {
+      amazonProvided: true,
+    };
+  }
+
+}
+
+/**
+ * Represents an IPv4 address range managed by AWS IP Address Manager (IPAM).
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipam.html
+ */
+class IpamIpv6 implements IIpAddresses {
```

Comment:
I confirmed that if the class is not exported, you do not need to add `@internal` annotation. You can ignore my previous comments to add `@internal`